Bug 1768771: network attachment definition static models missing crd flag

spadgett opened the pull request:
@pcbailey These models should have the `crd: true` set since they're CRDs. Let me know if you have concerns on this.

/assign @cyril-ui-developer @pcbailey

spadgett commented:
Note that this will change the resource pages to include the full group/version/kind in the URL.

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/models/index.ts:
```diff
@@ -22,6 +23,7 @@ export const SriovNetworkNodePolicyModel: K8sKind = {
   abbr: 'SRNNPM', // TODO check on this
```
These abbreviations are too long. We'll want something shorter. Max is 4 characters, but we really try to keep to 3 when possible.

Definitely don't include the `M` since "model" is an internal console concept :)

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/models/index.ts:
```diff
@@ -34,4 +36,5 @@ export const HyperConvergedModel: K8sKind = {
   abbr: 'SRNNPM', // TODO check on this
```
This looks wrong :)

pcbailey replied:
haha... it is. I've already got a PR to fix it. https://github.com/openshift/console/pull/3202

spadgett commented:
I haven't changed the existing abbreviations, but we should fix those in a follow up.

cyril-ui-developer commented:
@spadgett 
The code change fixed the issue. 
Thanks,
![Screen Shot 2019-11-06 at 3 09 28 PM](https://user-images.githubusercontent.com/15249132/68334004-b38b2300-00a7-11ea-93a0-79f654eb9483.png)

pcbailey commented:
@spadgett When I apply these changes and attempt to access the NAD list page, I get a 404 error on the page and the following warning in the console: `kindObj: no model for kind NetworkAttachmentDefinition`.

Thanks for the information on the abbreviations. I'll definitely fix them in a follow up.

spadgett commented:
Let me look at that problem. Any chance you can share your cluster details privately so I can test?

pcbailey commented:
Sure. I'll send them to you on slack.

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx:
```diff
@@ -106,7 +111,7 @@ const createNetAttachDef = (
     .then(() => {
       setLoading(false);
-      history.push(`/k8s/ns/${namespace || 'default'}/networkattachmentdefinitions`);
+      history.push(resourcePathFromModel(NetworkAttachmentDefinitionModel, name, namespace));
```
We typically go to the resource details rather than the list when creating new resources.

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx:
```diff
@@ -288,14 +293,7 @@ const NetworkAttachmentDefinitionFormBase = (props) => {
-              type="button"
-              variant="secondary"
-            >
+            <Button id="cancel" onClick={history.goBack} type="button" variant="secondary">
```
We typically use `history.goBack` on cancel.

spadgett commented:
/hold
for @pcbailey to test (thanks!)

spadgett commented:
OK, updated. I've fixed the Search page now as well, which was broken for this resource. @pcbailey PTAL, thanks!

![Screen Shot 2019-11-07 at 9 55 24 AM](https://user-images.githubusercontent.com/1167259/68399712-d586b300-0144-11ea-8752-822dfbff415d.png)

![Screen Shot 2019-11-07 at 9 53 39 AM](https://user-images.githubusercontent.com/1167259/68399722-d9b2d080-0144-11ea-8a56-9851ea35d916.png)

![Screen Shot 2019-11-07 at 9 53 36 AM](https://user-images.githubusercontent.com/1167259/68399732-dd465780-0144-11ea-9ef0-afef3ab7faeb.png)

spadgett commented on frontend/packages/kubevirt-plugin/src/utils/index.ts:
```diff
@@ -139,6 +140,9 @@ export const getResource = (
+    //
+    // FIXME: This is incorrect! `m.kind` is not unique. These model definitions should have `crd: true`, which will
+    // break this utility. We should be using `referenceForModel` and `crd: true` in our model definitions!
     kind: m.kind,
```
@vojtechszocs FYI, this is a problem... (see my comment). It looks like we might be doing this throughout kubevirt, and it will cause weird bugs in different parts of the UI like https://bugzilla.redhat.com/show_bug.cgi?id=1768771 and the broken search page for NetworkAttachmentDefinition.

I'll probably open a bug to look at this, but the change likely won't be small :(

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionDetails.tsx:
```diff
@@ -62,22 +64,20 @@ export const NetAttachDefinitionSummary: React.FC<NetAttachDefinitionSummaryProp
-    <div className="co-m-pane__body">
-      <StatusBox data={netAttachDef} loaded={!!netAttachDef}>
-        <ScrollToTopOnMount />
-        <div className="co-m-pane__body">
```
This was causing extra padding in the details page.

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/plugin.ts:
```diff
@@ -91,27 +92,13 @@ const plugin: Plugin<ConsumedExtensions> = [
-      exact: true,
-      path: [
-        '/k8s/ns/:ns/networkattachmentdefinitions',
-        '/k8s/all-namespaces/networkattachmentdefinitions',
```
This is unnecessary.

spadgett commented on frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionDetailsPage.tsx:
```diff
@@ -15,18 +15,6 @@ const menuActions = [
 > = (props) => {
-  const { name, namespace } = props;
-
-  const resources = [
```
This isn't needed. The resource will already be watched for details pages. This will watch it twice.

spadgett commented:
/hold cancel

pcbailey commented:
@spadgett Everything I tested worked fine for me except for network type dropdown options. It's because I'm using `model.kind` in the `mapStateToProps` function for `NetworkAttachmentDefinitionForm`. You can fix it in this PR if you want, or I can fix it in a follow up. Just let me know how you want to handle it.

pcbailey commented:
/lgtm

pcbailey commented:
/lgtm

spadgett commented:
Fixed the lint error

spadgett commented:
> level=fatal msg="failed to fetch Terraform Variables: failed to fetch dependency of \"Terraform Variables\": failed to fetch dependency of \"Bootstrap Ignition Config\": failed to generate asset \"Master Machines\": failed to fetch availability zones: failed to list zones: Get https://www.googleapis.com/compute/v1/projects/openshift-gce-devel-ci/zones?alt=json&filter=%28region+eq+https%3A%2F%2Fwww.googleapis.com%2Fcompute%2Fv1%2Fprojects%2Fopenshift-gce-devel-ci%2Fregions%2Fus-east1%29+%28status+eq+UP%29&prettyPrint=false: oauth2: cannot fetch token: Post https://oauth2.googleapis.com/token: dial tcp: i/o timeout"

/retest

pcbailey commented:
/lgtm
